### PR TITLE
[ Ent-Corda ] Ansible role to create namespace, serviceaccounts and Vault access

### PR DESCRIPTION
**Pull request by himanshulalarya**
- Added Ansible role to create namespace, serviceaccounts, storage class and Vault access for corda enterprise edition

 

**Reviewed by**
@sownak @jagpreetsinghsasan @deepakkumardbd 

 

**Linked issue**
#799

### CODE REVIEW

File: platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: {{ component_name }}
+provisioner: kubernetes.io/aws-ebs
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+parameters:
+  encrypted: "true"
```
Comment (deepakkumardbd) [inline platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl:9]: new line

File: platforms/corda-ent/configuration/create/k8_component/templates/mini_storageclass.tpl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: {{ component_name }}
+provisioner: k8s.io/minikube-hostpath
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+parameters:
+  encrypted: "true"
```
Comment (deepakkumardbd) [inline platforms/corda-ent/configuration/create/k8_component/templates/mini_storageclass.tpl:9]: add new line in all files at the end

File: platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl (hunk 1 above)
Comment (himanshulalarya) [reply platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl]: Done

File: platforms/corda-ent/configuration/create/k8_component/templates/mini_storageclass.tpl (hunk 2 above)
Comment (himanshulalarya) [reply platforms/corda-ent/configuration/create/k8_component/templates/mini_storageclass.tpl]: Done

File: platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    cd ./build    
+    echo "path \"secret/{{ component_ns }}/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }
+    path \"secret/cenm/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }
+    path \"secret/notary/*\" {
+        capabilities = [\"read\", \"list\", \"create\"]
+    }"  >>  vault-crypto-{{ component_type }}-{{ component_name }}-ro.hcl
```
Comment (deepakkumardbd) [inline platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml:98]: Will it not enable the path for all the org ?

Comment (jagpreetsinghsasan) [reply platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml:98]: Separate policies for each organization

File: platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (deepakkumardbd) [inline platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl:10]: new line

File: platforms/hyperledger-fabric/releases/dev/carrier-ent/reviewer_rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: carrier-ent-role-tokenreview-binding
+  namespace: default
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
```
Comment (deepakkumardbd) [inline platforms/hyperledger-fabric/releases/dev/carrier-ent/reviewer_rbac.yaml:8]: Releases files should not be checked-in. why is it in Fabric one?

File: platforms/corda-ent/configuration/create/k8_component/vars/main.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+dlt_templates:
+  doorman: create_doorman.tpl
```
Comment (sownak) [inline platforms/corda-ent/configuration/create/k8_component/vars/main.yaml:2]: remove the tpl references that are not used.

File: platforms/corda-ent/configuration/samples/network-cordaent.yaml [hunk 7]
```diff
@@ -4,7 +4,7 @@
 network:
   # Network level configuration specifies the attributes required for each organization
   # to join an existing network.
-  type: corda
+  type: cordaent
```
[the remark this answers is not in the document]
Comment (himanshulalarya) [reply platforms/corda-ent/configuration/samples/network-cordaent.yaml]: Done

File: platforms/shared/configuration/site.yaml [hunk 8]
```diff
@@ -24,6 +24,8 @@
     vars:
       add_new_org: false
     when: network.type == 'quorum' and (reset is undefined or reset == 'false')
+  - import_playbook: "{{ playbook_dir }}/../../corda-ent/configuration/deploy-network.yaml"
+    when: network.type == 'cordaent' and (reset is undefined or reset == 'false')
```
Comment (sownak) [inline platforms/shared/configuration/site.yaml:28]: corda-enterprise

Comment (himanshulalarya) [reply platforms/shared/configuration/site.yaml]: changed

File: platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl (hunk 4 above)
Comment (sownak) [reply platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl:10]: this comment is not addressed

File: platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml (hunk 3 above)
Comment (sownak) [reply platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml:119]: Is this conv resolved?

Comment (himanshulalarya) [reply platforms/corda-ent/configuration/setup/vault_kubernetes/tasks/main.yaml:119]: Yes, It's already resolved

File: platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl (hunk 4 above)
Comment (himanshulalarya) [reply platforms/corda-ent/configuration/create/k8_component/templates/eks_storageclass.tpl:10]: It addressed. New line was added

**Pull request merged**